test/document.test.js

```diff
@@ -10805,7 +10805,10 @@ describe('document', function() {
     assert.ok(!band.populated('members'));
     assert.ok(!band.populated('lead'));
     assert.ok(!band.populated('embeddedMembers.member'));
+    assert.ok(band.members.isMongooseArray);
+    assert.ok(band.embeddedMembers.isMongooseArray);
     assert.ok(!band.embeddedMembers[0].member.name);
+    // assert.ok(!band.embeddedMembers[0].$populated('member'));
```

Comment:
> I left assert.ok(!band.embeddedMembers[0].$populated('member')); commented out because that assertion fails, but I want to look at it some more when we come back to https://github.com/Automattic/mongoose/issues/1635.

was this working before the PR?

Reply:
I confirmed this assertion fails before this PR. We will need to do some extra work to make sure `depopulate()` also updates subdocs' `$populated()`.